=== modified file 'libraries/libapparmor/swig/SWIG/libapparmor.i'
--- libraries/libapparmor/swig/SWIG/libapparmor.i	2014-01-06 22:08:55 +0000
+++ libraries/libapparmor/swig/SWIG/libapparmor.i	2014-03-07 19:54:12 +0000
@@ -9,6 +9,16 @@
 %include "typemaps.i"
 %include <aalogparse.h>
 
+#ifdef SWIGPYTHON
+%exception {
+  $action
+  if (result < 0) {
+    PyErr_SetFromErrno(PyExc_OSError);
+    return NULL;
+  }
+}
+#endif
+
 /* swig doesn't like the macro magic we do in apparmor.h so the fn prototypes
  * are manually inserted here
  */
@@ -37,3 +37,5 @@
 extern int aa_getcon(char **con, char **mode);
 extern int aa_getpeercon_raw(int fd, char *buffer, int *size);
 extern int aa_getpeercon(int fd, char **con);
+
+%exception;

